pkg/fileutil/fileutil.go

```diff
@@ -238,3 +238,14 @@ func IsExecutable(path string) bool {
 	}
 	return false
 }
+
+// IsDeviceNode checks if the given path points to a block or char device.
+// It doesn't follow symlinks.
+func IsDeviceNode(path string) bool {
+	d, err := os.Lstat(path)
```

Comment:
Is it correct not to follow symlinks?
```
$ ls -l /dev/rtc*
lrwxrwxrwx. 1 root root      4 Jul  8 17:02 /dev/rtc -> rtc0
crw-------. 1 root root 251, 0 Jul  8 17:02 /dev/rtc0
```
With `/dev/rtc0`, this works:
```
$ sudo rkt run --debug  --insecure-options=image,ondisk --interactive --volume=disk,kind=host,source=/dev/rtc0 docker://busybox --mount=volume=disk,target=/opt/the-dev --exec /bin/cat -- /opt/the-dev
```
With `/dev/rtc`, this does not work:
```
$ sudo rkt run --debug  --insecure-options=image,ondisk --interactive --volume=disk,kind=host,source=/dev/rtc docker://busybox --mount=volume=disk,target=/opt/the-dev --exec /bin/cat -- /opt/the-dev
...
cat: can't open '/opt/the-dev': Operation not permitted
```

The mounting logic of nspawn follows links, so I think the `DeviceAllow=` logic should follow links in the same way.